packages/adapter/addon/rest.js

```diff
@@ -309,18 +310,47 @@ const RESTAdapter = Adapter.extend(BuildURLMixin, {
     },
   }),
 
-  useFetch: computed(function() {
-    let ENV = getOwner(this).resolveRegistration('config:environment');
-    // TODO: https://github.com/emberjs/data/issues/6093
-    let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
-
-    if (jQueryIntegrationDisabled) {
-      return true;
-    } else if (hasNajax || hasJQuery) {
-      return false;
-    } else {
-      return true;
-    }
+  useFetch: computed({
+    get() {
+      if (this._useFetch !== undefined) {
+        return this._useFetch;
+      }
+      let ENV = getOwner(this).resolveRegistration('config:environment');
+      // TODO: https://github.com/emberjs/data/issues/6093
+      let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
+
+      if (jQueryIntegrationDisabled) {
+        return (this._useFetch = true);
+      } else if (DEPRECATE_NAJAX && typeof najax !== 'undefined') {
+        if (has('fetch')) {
+          deprecate(
+            'You have ember-fetch and jquery installed. To use ember-fetch instead of najax, set `useFetch: true` in your adapter.  In 4.0, ember-data will default to ember-fetch instead of najax when both ember-fetch and jquery are installed in FastBoot.',
```

Comment:
Can we tweak this to be "native class" first (e.g. instead of `useFetch: true` say `useFetch = true`)?